x-pack/legacy/plugins/epm/server/lib/kibana/index_pattern/install.ts

```diff
@@ -102,7 +102,39 @@ export const dedupeFields = (fields: Fields) => {
   return Object.values(uniqueObj);
 };
 
-export const transformField = (field: Field): IndexPatternField => {
+/**
+ * search through fields with field's path property
+ * returns undefined if field not found or field is not a leaf node
+ * @param  allFields fields to search
+ * @param  path dot separated path from field.path
+ */
+export const findFieldByPath = (allFields: Fields, path: string): Field | undefined => {
+  const pathParts = path.split('.');
+  const getField = (fields: Fields, pathNames: string[]): Field | undefined => {
```

Comment:
I think I would have this function on the same level as the others, just not export it if it isn't used elsewhere.

Reply:
I'm only exporting it to use it in a test file

Reply:
misunderstood and agreed, will make the change in the next PR.